dlib/pixel.h

```diff
@@ -474,7 +474,9 @@ namespace dlib
     struct pixel_traits<rgb_pixel>
     {
         constexpr static bool rgb  = true;
+        constexpr static bool bgr  = false;
```

Comment:
I wasn't sure about adding these, but felt like a waste to not implement support for using the BGR decoding in libwebp, especially since opencv uses it (even if the rest of the necessary support hasn't been added, there's at least one less place to update in the future if it is implemented).

Reply:
Er, I just looked at the code. I see why the pixel traits are updated, looked on my phone earlier :)

Anyway.  Yeah this is fine.  Although rename `bgr` to `bgr_layout`.  You then also don't need the `bgr_alpha` field. Since `bgr_layout` is a modifier tells you everything that's needed, which is that the pixel's memory layout goes BGR(A) rather than RGB(A).

Reply:
And update the docs at the top of this file to explain the meaning of `bgr_layout` :)